From 5316dfe69daeb0c843f44c7e411582a785ffe505 Mon Sep 17 00:00:00 2001
From: Sriharsha Govinda Raj <sriharsha.govinda.raj@intel.com>
Date: Tue, 12 Apr 2016 13:11:07 +0530
Subject: [PATCH 2/5] XPWR/CHRG: DCP inlimit settings based on usb violation
 bit in OEM1 table.

Making a check on compliance bit and setting the DCP inlimit
to 1.5A in according to USB BC 1.2 compliance spec.

Change-Id: Idd048ace3faf0f088982aa41a8ed57afac5adeb4
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-24264
Signed-off-by: Sriharsha <sriharsha.govinda.raj@intel.com>
Reviewed-on: https://android.intel.com:443/493380
---
 drivers/extcon/extcon-dc-pwrsrc.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/extcon/extcon-dc-pwrsrc.c b/drivers/extcon/extcon-dc-pwrsrc.c
index 40265a8..a29009c 100644
--- a/drivers/extcon/extcon-dc-pwrsrc.c
+++ b/drivers/extcon/extcon-dc-pwrsrc.c
@@ -85,7 +85,8 @@
 #define DC_BC12_IRQ_CFG_REG		0x45
 #define BC12_IRQ_CFG_MASK		0x2
 
-#define DC_XPWR_CHARGE_CUR_DCP		2000
+#define DC_XPWR_CHARGE_CUR_DCP_1500		1500
+#define DC_XPWR_CHARGE_CUR_DCP_2000		2000
 #define DC_XPWR_CHARGE_CUR_CDP		1500
 #define DC_XPWR_CHARGE_CUR_SDP_500	500
 #define DC_XPWR_CHARGE_CUR_SDP_100	100
@@ -270,7 +271,10 @@ static int handle_chrg_det_event(struct dc_pwrsrc_info *info)
 		notify_charger = true;
 		cable_props.chrg_evt = POWER_SUPPLY_CHARGER_EVENT_CONNECT;
 		cable_props.chrg_type = POWER_SUPPLY_CHARGER_TYPE_USB_DCP;
-		cable_props.ma = DC_XPWR_CHARGE_CUR_DCP;
+		if (info->pdata->chrg_usb_compliance)
+			cable_props.ma = DC_XPWR_CHARGE_CUR_DCP_1500;
+		else
+			cable_props.ma = DC_XPWR_CHARGE_CUR_DCP_2000;
 	} else {
 		dev_warn(&info->pdev->dev,
 			"disconnect or unknown or ID event\n");
-- 
1.9.1

